src/main/java/umaikaze/duke/TaskList.java

```diff
@@ -0,0 +1,97 @@
+package umaikaze.duke;
+
+import umaikaze.duke.task.Deadline;
+import umaikaze.duke.task.Event;
+import umaikaze.duke.task.Task;
+import umaikaze.duke.task.Todo;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class TaskList {
+    List<Task> list;
+
+    public TaskList() {
+        list = new ArrayList<>(100);
+    }
+
+    public TaskList(List<Task> loaded) {
+        list = loaded;
+    }
+
+    public String countList() {
+        return "Nyow you have " + list.size() + " tasks in the wist.";
+    }
+
+    private Task getTask(String[] line) throws DukeException{
+        Task newTask;
+        String cmd = line[0];
+        Parser p = new Parser(line);
+        switch (cmd) {
+            case "deadline":
+                if (p.getDescription().equals("")) {
+                    throw new DukeException("OOPS ;;ω;;  The descwiption of a deadwinye cannyot be empty.");
+                }
+                if (p.time == null) {
+                    throw new DukeException("OOPS ;;ω;;  The deadwinye of a deadwinye cannyot be empty" +
+                            ", did you use /by to state the deadwinye?");
+                }
+                newTask = new Deadline(p.getDescription(), p.time, p.hasTime);
+                break;
+            case "event":
+                if (p.getDescription().equals("")) {
+                    throw new DukeException("OOPS owo  The descwiption of a event cannyot be empty.");
+                }
+                if (p.time == null) {
+                    throw new DukeException("OOPS ;;ω;;  The timing fow an event cannyot be empty," +
+                            " did you use /at to state the timing?");
+                }
+                newTask = new Event(p.getDescription(), p.time, p.hasTime);
+                break;
+            case "todo":
+                if (p.getDescription().equals("")) {
+                    throw new DukeException("OOPS (・`ω´・)  The descwiption of a todo cannyot be empty.");
+                }
+                newTask = new Todo(p.getDescription());
+                break;
+            default:
+                throw new DukeException("OOPS oωo  I'm sowwy, but I don't knyow what that means ^;;ω;;^");
+        }
+        return newTask;
+    }
+
+    public String addTask(String[] line) throws DukeException{
+        Task newTask = getTask(line);
+        list.add(newTask);
+        return "Got it ^UωU^ I've added this task: \n\t"
```

Comment:
Similarly, you might want to make `ui` `showAddingTaskMessage()` to the users. 😃